File: frontend/src/components/map/junctions.js
const junctions = [
  {
    id: 1,
    name: "Junction A",
    position: [13.0827, 80.2707],
    load: 92,
    greenTime: 30,
    neighbors: [2],
  },
  {
    id: 2,
    name: "Junction B",
    position: [13.0875, 80.2780],
    load: 42,
    greenTime: 38,
    neighbors: [1, 3],
  },
  {
    id: 3,
    name: "Junction C",
    position: [13.0755, 80.2635],
    load: 67,
    greenTime: 30,
    neighbors: [2],
  },
];

export default junctions;

File: backend/app/routes/junctions.py
from fastapi import APIRouter, Depends, HTTPException
from sqlalchemy.orm import Session

from app.database.dependency import get_db
from app.database.models import Junction

router = APIRouter()


# --------------------------------------------------
# GET ALL JUNCTIONS
# --------------------------------------------------

@router.get("/")
def get_junctions(db: Session = Depends(get_db)):
    return db.query(Junction).all()


# --------------------------------------------------
# UPDATE JUNCTION
# --------------------------------------------------

@router.put("/{junction_id}")
def update_junction(
    junction_id: int,
    load: int,
    averageSpeed: int,
    greenTime: int,
    db: Session = Depends(get_db),
):

    junction = (
        db.query(Junction)
        .filter(Junction.id == junction_id)
        .first()
    )

    if junction is None:
        raise HTTPException(
            status_code=404,
            detail="Junction not found"
        )

    junction.load = load
    junction.averageSpeed = averageSpeed
    junction.greenTime = greenTime

    db.commit()
    db.refresh(junction)

    return junction